algorithms/CPlusPlus/Arrays/reverse-array.cpp

```diff
@@ -0,0 +1,31 @@
+//Description : This program reverses array elements using two pointer method.
+//Time Complexity : O(n), where n is the array size
+//Auxiliary Space : O(1)
+#include<bits/stdc++.h>
+using namespace std ;
+void reverse_array(int *arr, int n){
+// Here we will use two pointer approach
+int low = 0 ;     //First pointer at the beginning of the array
+int high = n-1;  //Second pointer at the end of the array
+
+while(low < high){
+  //Swapping array items at the low and high index
+  swap(arr[low],arr[high]);
+  low++;  //Incrementing the low pointer
+  high--; //Decrementing the high pointer
+}
+
+  for(int i = 0; i < n ; i++){
+  cout<<arr[i]<<" ";
+  }
+}
+int main(){
+  int n ; //Size of the array
+  cin>>n;
+  int arr[n]; //Array of size n
```

Comment:
use of VLAs (Variable Size Arrays) is not allowed in C++. Although some compilers do support it ( as non-standard extension ) but using it ( especially like this ) could lead to a potential risk of having a massive array on your stack memory, leaving little to no space for rest of the program.

Since you have opted for C++, it's much better to go with C++ idioms like `std::vectors` in this place instead of going for any C idioms in your program until necessary.